honeycomb/honeycomb.go

```diff
@@ -423,6 +423,19 @@ func NewExporter(config Config, opts ...ExporterOption) (*Exporter, error) {
 	}, nil
 }
 
+// Consume from the response queue, calling the onError callback when errors are encountered.
+func (e *Exporter) StartErrorLogger() {
+	responses := libhoney.TxResponses()
+	go func() {
+		for {
+			r, ok := <-responses
+			if ok && r.Err != nil {
+				e.onError(r.Err)
+			}
+		}
```

Comment:
should there be a shutdown hook here?

Reply:
We could accept a `context.Context` parameter and honor its `Done` channel, or accept a readable `<-chan struct{}` that we expect to become ready to indicate that we should stop. Rather than blocking on reading only from the `responses` channel, we could then use `select` to also notice when we should exit the loop.

I'd prefer that this function operate _synchronously_, and allow callers to spawn it in a separate goroutine if they wish to. That is, leave concurrency in the hands of the caller.

Reply:
Thank you both. Yeah, I like the idea of controlling via a `context.Context` and running synchronously. Let me push that change and ask you both to re-review.

Reply:
Notable that this will be run from a daemon (collector) and services using the SDK, so important to get right I think.